### PR TITLE
Routing/quiz course

### PR DESCRIPTION
**Pull request by Nesrine1333**
removed the go back button , I haven't finished it yet, there's no direct routre for the paid course 
changed the api form https://127.1... to the one with .env (quiz component)
signup shouldn't be double 
home footer the new one (the deployed one is the old )
the colose popup is still not stable

### CODE REVIEW

File: src/component/Quiz/nabar/index.jsx [hunk 1]
```diff
@@ -21,7 +47,7 @@ const QuizNav = () => {
     </div>
    <div  className={styles.text}>Assess Your Knowledge Of The Product Owner Accountabilities</div> 
 </div>
-<div> <button>Go Back</button></div>
+{/* <div> <button onClick={handlehome}>Go Back</button></div> */}
```
Comment (TomIA24) [inline src/component/Quiz/nabar/index.jsx:50]: this button did not handle any action, why ?

**Review by TomIA24 — CHANGES_REQUESTED** [review]

File: src/component/Course/index.jsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import PaidCourse from "./paid"
 const  Course = () =>{
 
     let { id } = useParams();
+    console.log('Course ID:', id);
```
Comment (elayeboussama) [inline src/component/Course/index.jsx:10]: don't forget to remove console.log after testing

File: src/component/Nav/index.jsx [hunk 3]
```diff
@@ -49,6 +50,27 @@ const Nav = () => {
...
+
+  const [candidateData, setcandidateData] = useState([]);
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+
+        const candidateResponse = await axios.get(`${process.env.REACT_APP_API}api/candidat/candidates/${user._id}`);
+        const candidateData = candidateResponse.data;
+   
+        setcandidateData(candidateResponse.data);
```
Comment (elayeboussama) [inline src/component/Nav/index.jsx:63]: change it to setCandidateData

File: src/component/Nav/index.jsx [hunk 4]
```diff
@@ -61,7 +83,11 @@ const Nav = () => {
   };
 
   const closepopup = () => {
-    navigate("/candidate/profile");
+    console.log('close popup');
```
Comment (elayeboussama) [inline src/component/Nav/index.jsx:86]: don't forget to remove console.log after testing

File: src/component/Nav/index.jsx [hunk 5]
```diff
@@ -61,7 +83,11 @@ const Nav = () => {
...
+    
+    setTimeout(() => {
+      navigate('/candidate/profile'); // Navigate after state update
+    }, 100);
     setpopupopen(false);
```
Comment (elayeboussama) [inline src/component/Nav/index.jsx:91]: change it to setPopupOpen

File: src/component/Nav/index.jsx [hunk 6]
```diff
@@ -444,7 +470,7 @@ const Nav = () => {
               <div className={styles.closbutton}>
                 {" "}
```
Comment (elayeboussama) [inline src/component/Nav/index.jsx:471]: remove this

File: src/component/Personalize/index.jsx [hunk 7]
```diff
@@ -1022,7 +1022,8 @@ console.log(e.target.value);
       console.log("Data being sent:", stepData);
 
       const response = await axios.put(apiUrl, stepData);
-
+      const candidateResponse = await axios.get(`${process.env.REACT_APP_API}api/candidat/candidates/${candiddId}`);
+      setcandidatdata(candidateResponse.data)
```
Comment (elayeboussama) [inline src/component/Personalize/index.jsx:1026]: change this to setCandidateData

File: src/component/Quiz/Questions/index.jsx [hunk 8]
```diff
@@ -506,53 +514,53 @@ const Quiz = ({ startDate }) => {
...
-                                />
-                                <span className={styles.checkmark}></span>
-                                {answer}
-                              </label>
-                            ))}
+                          .concat(questions[currentQuestion].wrongAnswers)
+                          .map((answer, index) => (
+                            <label
+                              key={index}
+                              className={styles.checkmarkcontainer}
```
Comment (elayeboussama) [inline src/component/Quiz/Questions/index.jsx:542]: camel case naming

File: src/component/Quiz/Questions/index.jsx [hunk 9]
```diff
@@ -506,53 +514,53 @@ const Quiz = ({ startDate }) => {
...
-                                />
-                                {answer}
-                                <span className={styles.radio}></span>
-                              </label>
-                            ))
+                          .concat(questions[currentQuestion].wrongAnswers)
+                          .map((answer, index) => (
+                            <label
+                              key={index}
+                              className={styles.rdiocontainer}
```
Comment (elayeboussama) [inline src/component/Quiz/Questions/index.jsx:521]: camel case naming

File: src/component/Quiz/Questions/index.jsx [hunk 10]
```diff
@@ -506,53 +514,53 @@ const Quiz = ({ startDate }) => {
...
+                                  )
+                                }
+                                onChange={() =>
+                                  handleCheckboxChange(
+                                    currentQuestion,
+                                    answer
+                                  )
+                                }
+                              />
+                              <span className={styles.checkmark}></span>
```
Comment (elayeboussama) [inline src/component/Quiz/Questions/index.jsx:560]: camel case naming

File: src/component/Quiz/nabar/index.jsx [hunk 11]
```diff
@@ -2,10 +2,49 @@ import React, { useState, useRef,useEffect } from 'react';
 import axios from 'axios';
 import styles from "./styles.module.css";
 import Nav from "../../Nav";
+import { Link, useNavigate, useLocation } from "react-router-dom"; 
 
-const QuizNav = () => {
+const QuizNav = ({quizId}) => {
+console.log("quizId",quizId);
```
Comment (elayeboussama) [inline src/component/Quiz/nabar/index.jsx:8]: don't forget to remove console.log after testing

File: src/component/Quiz/nabar/index.jsx [hunk 12]
```diff
@@ -2,10 +2,49 @@ import React, { useState, useRef,useEffect } from 'react';
...
+
+
+    };
+    fetchQuiz();
+  }, [quizId]);
+  console.log(`${process.env.REACT_APP_API}api/quiz/${quizId}`);
+  const navigate = useNavigate();
+
+function goToHome() {
+  console.log("quiz.courseID",quiz.courseID);
```
Comment (elayeboussama) [inline src/component/Quiz/nabar/index.jsx:43]: don't forget to remove console.log after testing

File: src/component/Quiz/nabar/styles.module.css [hunk 13]
```diff
@@ -6,6 +6,16 @@
     z-index: -1; /* Ensure the background is behind the content */
   }
   
+  .pdcontainer{
```
Comment (elayeboussama) [inline src/component/Quiz/nabar/styles.module.css:9]: camel case naming

File: src/component/Quiz/nabar/styles.module.css [hunk 14]
```diff
@@ -6,6 +6,16 @@
...
+    display: flex;
+    flex-direction: row;
+    align-items: flex-end;
+    justify-content: space-between;
+    padding-left: 12%;
+    padding-right: 12%;
+  }
+
+
   .imagebackground {
```
Comment (elayeboussama) [inline src/component/Quiz/nabar/styles.module.css:19]: camel case naming

File: src/component/Quiz/nabar/styles.module.css [hunk 15]
```diff
@@ -42,8 +52,8 @@ font-weight: bold;
     margin-top: 2em;
   }
   .pdowncontainer{
-    padding-left: 13.5%;
-    padding-right: 13.5%;
+    /* padding-left: 13.5%;
+    padding-right: 13.5%; */
```
Comment (elayeboussama) [inline src/component/Quiz/nabar/styles.module.css:56]: Remove unnecessary comments

**Review by elayeboussama — CHANGES_REQUESTED** [review]

File: src/component/Quiz/nabar/index.jsx (hunk 1 above)
Comment (Nesrine1333) [reply src/component/Quiz/nabar/index.jsx]: Done

File: src/component/Nav/index.jsx (hunk 5 above)
Comment (Nesrine1333) [reply src/component/Nav/index.jsx:91]: done

File: src/component/Nav/index.jsx (hunk 6 above)
Comment (Nesrine1333) [reply src/component/Nav/index.jsx:471]: Done

File: src/component/Quiz/nabar/styles.module.css (hunk 15 above)
Comment (Nesrine1333) [reply src/component/Quiz/nabar/styles.module.css:56]: done

File: src/component/Nav/index.jsx (hunk 3 above)
Comment (Nesrine1333) [reply src/component/Nav/index.jsx:63]: done

**Pull request merged**